### PR TITLE
CLI foundations

### PR DESCRIPTION
**Pull request by iostat**
`npm i -g` ;)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,9 @@ dist: trusty
 sudo: required
 node_js: stable
 install:
-  - npm install
-  - bower install --production
+- npm install -g bower pulp purescript@0.11.7
```
Comment (safareli) [inline .travis.yml]: `npm i` will install this libs too so there should be no need for this.
also in travis, binaries of all installed npm libs is available in PATH

File: package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "preinstall": "bower install --production && pulp build -t chanterelle.js",
```
Comment (safareli) [inline package.json]: I think you need `postinstall`

**Comment by iostat** [comment]
yeah i think postinstall will allow me to not have to install bower, etc.
beforehand. cheers!

**Comment by blinky3713** [comment]
is this done?

**Pull request merged**